src/main/java/com/commercetools/sync/producttypes/helpers/AttributeDefinitionReferenceResolver.java

```diff
@@ -112,12 +112,12 @@ public AttributeDefinitionReferenceResolver(@Nonnull final ProductTypeSyncOption
         final String resourceKey;
         try {
             resourceKey = getKeyFromResourceIdentifier(typeReference);
-        } catch (ReferenceResolutionException exception) {
+            return productTypeService.fetchCachedProductTypeId(resourceKey)
+                    .thenApply(optionalId -> optionalId.map(ProductType::referenceOfId));
+        } catch (Exception exception) {
```

Comment:
Same as above.

Reply:
These changes has been obsoleted. Please refer to followings
https://github.com/commercetools/commercetools-sync-java/commit/b6d96a84e6d2ebcf370f8145904cdecc894b8cd2